torch/nested/_internal/ops.py

```diff
@@ -724,6 +724,53 @@ def chunk_default(func, *args, **kwargs):
         ]
 
 
+@register_jagged_func(
+    torch.ops.aten.chunk_backward.default,
+    "grads: any, input: jt, chunks: any, dim: any",
+)
+def chunk_backward(func, *args, **kwargs):
+    _, new_kwargs = normalize_function(
+        func, args=args, kwargs=kwargs, normalize_to_only_use_kwargs=True
+    )
+    grads = new_kwargs.pop("grads")
+    inp = new_kwargs.pop("input")
+
+    # Only support the simple case when dim != 0 for now
+    # To support chunk_backward dim=0, need to support cat(dim=0) first
+    dim = _wrap_jagged_dim(
+        inp.dim(), new_kwargs["dim"], "chunk_backward", convert_to_inner_dim=False
+    )
+
+    need_to_fill_grad = False
+    for grad in grads:
+        if grad is None:
+            need_to_fill_grad = True
+            break
+
+    if need_to_fill_grad:
+        # Since torch.zeros([.., j0, ..]) is not supported, skipping this case for now
+        raise RuntimeError(
+            "NestedTensor chunk backward requires that all grads are not NONE"
+        )
```

Comment:
The issue is that saving just shape is not enough because the shape itself does not contain the raggedness information today. We'd be able to easily support this later when we store the offsets on shape as part of the factory function support stack, but it is tricky today.